### DISCUSSION TITLE
Repel keeps saying I didn't enter a valid token

### DISCUSSION THREAD

**Discussion by LarchyEverything** [marioparaschiv/nitro-sniper#378] [Q&A] [answered]
I entered the tokens where I'm supposed to put it and it just says "All tokens provided were invalid, exiting process...
exit status 255". I may be setting up the settings wrong but nothing says I am. I copied all the code in https://github.com/localip/nitro-sniper/wiki/Default-Configuration and I entered the tokens right in between 'these things'. I'm not a coder so I may be oblivious to something that's obvious. I'm just unsure what I'm doing wrong and the previous question that was basically answering the same thing was filled with people arguing over if this is a token grabber. I don't care if it is or isn't. Just help me fix this issue.

**Comment by mrxiaoman** [comment]
Share your settings

**Reply by LarchyEverything** [reply]
I put actual account tokens where the 'token' is, I just don't wanna reveal my tokens.

{
   tokens: {
      // Main Token (ex: Nz...)
      main: 'TOKEN',
      // Alt Tokens (ex: Nz...)
      alts: [
         'TOKEN',
      ],
   },
   // The mode to run the sniper in. Options: main (only main account), alts (only alts), both
   mode: 'alts',
   /*
      The status the accounts should be on.
      Options: online, dnd, idle, offline, default
      default means the status will not be modified.
   */
   status: {
      // The status the main account will have IF it's logged in
      main: 'default',
      // The status the logged in alts will have
      alts: 'default'
   },
   nitro: {
      // The amount of nitros needed to be sniped for the cooldown to activate
      max: 2,
      // Cooldown to activate after max nitro has been hit (in hours)
      cooldown: 24,
      dm: {
         /*
          * How long to wait before attempting to claim the nitro in DMs (in seconds)
          * This is to avoid suspicion when winning giveaways by slowing down the sniper from instantly sniping it.
          */
         delay: 10
      },
      // A list of channel IDs to ignore when sniping nitro
      blacklistedChannels: [
         '',
      ]
   },
   giveaway: {
      // Wether or not to activate the giveaway sniper (true/false)
      enabled: true,
      // Delay to react to the giveaway (in seconds)
      delay: 30,
      // DM the hoster on giveaway win (true/false)
      dm: true,
      // Messages to DM the host
      dmMessages: [
         'Hey, i won the giveaway.',
         'Could i redeem my prize?'
      ],
      // How long to wait to DM (in seconds)
      dmDelay: 25,
      // How long to wait between messages (in seconds)
      messageDelay: 2,
      // Blacklisted words for giveaway prizes
      blacklistedWords: [
         'bot',
         'test',
         'ban'
      ],
      // Only react to whitelisted giveaway prizes (true/false)
      whitelistOnly: false,
      // Whitelisted words for giveaway prizes
      whitelistedWords: [
         'nitro'
      ],
      // Blacklisted Server IDs to not snipe giveaways on
      blacklistedServers: [
         ''
      ],
      // Only snipe giveaways on whitelisted servers (true/false)
      whitelistServersOnly: false,
      whitelistedServers: [
        ''
      ],
      prizeLength: {
         // The minimum character length of the prize (null = infinite)
         min: 4,
         // The maximum character length of the prize (null = infinite)
         max: null
      }
   },
   invite: {
      // Wether or not to activate the invite sniper (true/false)
      enabled: false,
      delay: {
         // Minimum delay to join the server (in seconds)
         min: 10,
         // Maximum delay to join the server (in seconds)
         max: 20
      },
      members: {
         // The minimum member count the server should have
         min: 1500,
         // The maximum member count the server should have
         max: 50000
      },
      // The amount of joined invites needed for the cooldown to activate
      max: 10,
      // Cooldown to activate after max joined invites has been hit (in hours)
      cooldown: 6,
      // Whether to accept an invite on any other alt/account (depends on "mode") if the current alt has max servers (true/false)
      queue: true,
      // Whether to allow the main token to snipe invites ONLY if sniper mode is set to "both" (true/false)
      onlyAlts: true
   },
   webhook: {
      // URL to fire webhook to for notifications (ex: https://discord.com/api/webhooks/.../...)
      url: 'URL',
      enabled: {
         // Fire webhook on invalid code (true/false)
         codeInvalid: true,
         // Fire webhook on already redeemed code (true/false)
         codeAlreadyRedeemed: true,
         // Fire webhook on sniped code (true/false)
         codeSuccess: true,
         // Fire webhook on giveaway enter (true/false)
         giveawayEntered: true,
         // Fire webhook on giveaway win (true/false)
         giveawayWin: true,
         // Fire webhook on invite join (true/false)
         inviteJoin: false,
         // Fire webhook on failure of sniping invite (true/false)
         inviteFail: false
      },
      mentionEveryone: {
         // Mention on invalid code (true/false)
         codeInvalid: false,
         // Mention on already redeemed code (true/false)
         codeAlreadyRedeemed: false,
         // Mention on sniped code (true/false)
         codeSuccess: true,
         // Mention on giveaway enter (true/false)
         giveawayEntered: false,
         // Mention on giveaway win (true/false)
         giveawayWin: true,
         // Mention on invite join (true/false)
         inviteJoin: false,
         // Mention on failure of sniping invite (true/false)
         inviteFail: false
      }
   },
   // Whether to log all console messages to file (true/false)
   log: true
}

**Comment by marioparaschiv** [comment, marked as answer]
Use https://jsonformatter.org/json5-validator to validate your config.
